pure-swift-notes/pure-swift-notes-concurrency.md

```diff
@@ -0,0 +1,324 @@
+### Concurrency Model
+
+SQLDelight's concurrency model is built around thread-local connections and explicit transaction management. For Swift, we can maintain similar safety guarantees while leveraging modern Swift concurrency features.
+
+#### Key Decisions
+
+1. **Transaction Isolation**
+
+```swift
+// Option A: Thread/Task-based isolation (like SQLDelight)
+class Database {
+    private let queue = DispatchQueue(label: "com.sqldelight.db")
+    
+    func transaction<T>(_ block: (Connection) throws -> T) throws -> T {
+        try queue.sync {
+            // Execute transaction
+        }
+    }
+}
+
+// Option B: Actor-based isolation
```

Comment:
its gotta be this one. dispatch queues are the older non-coroutine-style API

Reply:
Heard! Thank you